src/Render/MenuRenderer.php

```diff
@@ -0,0 +1,177 @@
+<?php declare(strict_types=1);
+
+namespace Torr\MenuBundle\Render;
+
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Symfony\Contracts\Translation\TranslatableInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+use Torr\HtmlBuilder\Builder\HtmlBuilder;
+use Torr\HtmlBuilder\Node\HtmlElement;
+use Torr\MenuBundle\Exception\MissingDependencyException;
+use Torr\MenuBundle\Item\Data\ActiveState;
+use Torr\MenuBundle\Item\MenuItem;
+use Torr\MenuBundle\Item\ResolvedMenuItem;
+use Torr\MenuBundle\Render\Options\RenderOptions;
+use Torr\MenuBundle\Resolver\MenuResolver;
+use Torr\Rad\Route\LinkableInterface;
+
+class MenuRenderer
+{
+	/**
+	 * @param iterable<ItemRenderVisitorInterface> $renderVisitors
+	 */
+	public function __construct(
+		private readonly MenuResolver $menuResolver,
+		private readonly ?UrlGeneratorInterface $urlGenerator = null,
+		private readonly ?TranslatorInterface $translator = null,
+		private readonly iterable $renderVisitors = [],
+	) {}
+
+
+	/**
+	 * Renders the tree from the given root
+	 */
+	public function render (?MenuItem $root, ?RenderOptions $renderOptions = null) : string
+	{
+		if (null === $root)
+		{
+			return "";
+		}
+
+		$renderOptions ??= new RenderOptions();
+		$resolvedRoot = $this->menuResolver->resolveMenu($root);
```

Comment:
```suggestion
	public function render (
		?MenuItem $root,
		RenderOptions $renderOptions = new RenderOptions(),
	) : string
	{
		if (null === $root)
		{
			return "";
		}

		$resolvedRoot = $this->menuResolver->resolveMenu($root);
```

Since we're on PHP 8.1, we might as well initialize the argument directly: https://3v4l.org/iaZVX#v8.1.11

Or do we expect someone explicitly passing in `null` for the `$renderOptions` argument? In that case, we need to keep the previous version.

Reply:
No reason, can be changed.